lib/engine/game/g_steam_over_holland/game.rb

```diff
@@ -356,25 +363,43 @@ def non_floated_corporations
         end
 
         def check_distance(route, visits, train = nil)
-          super
-
           raise GameError, 'Route cannot begin/end in a town' if visits.first.town? || visits.last.town?
+
+          super
         end
 
         def revenue_for(route, stops)
-          super
+          revenue = super
 
-          raise GameError, 'Route visits same hex twice' if route.hexes.size != route.hexes.uniq.size
+          abilities(route.corporation, :hex_bonus).each do |ability|
+            stops.each do |stop|
+              next unless ability.hexes.include?(stop.hex.name)
+
+              revenue += 20
+            end
+          end
```

Comment:
Your code will work, but it would also be possible to write this without an iterator. In theory this might be more efficient code. Something like this (would need to be tested):
```suggestion
          abilities = abilities(route.corporation, :hex_bonus)
          return revenue if abilities.empty?
          
          bonus_hexes = abilities.flat_map(&:hexes)
          bonus = 20 * stops.map(&:hex).map(&:coordinates).intersect(bonus_hexes).size
          revenue + bonus
```

Reply:
sure, yes this works too. Done